Recruit-Training/Advanced-Recruit-Training/src/BatteryStateDisplayService.cpp

```diff
@@ -26,8 +26,20 @@ void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& ba
     batteryStateOfChargeService_.addData(batteryData);
 
     QTextStream(stdout) << "Voltage: " << batteryData.voltage
-        << " Current: " << batteryData.current
-        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed() << endl;
+                        << " Current: " << batteryData.current
+                        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed()<< endl;
+
+    QTime displayTime = batteryStateOfChargeService_.timeWhenChargedOrDepleted();
+
+    if(batteryStateOfChargeService_.isCharging())
+        QTextStream(stdout) << "Time until charged:";
+    else
+        QTextStream(stdout) << "Time until depleted:";
+
+    QTextStream(stdout) << "  Hours:" << (displayTime.hour() + batteryStateOfChargeService_.getRemainingHours())
+                        << "  Min:" << batteryStateOfChargeService_.timeWhenChargedOrDepleted().minute()
+                        << "  Sec:" << batteryStateOfChargeService_.timeWhenChargedOrDepleted().second()
+                        << "  Ms:" << batteryStateOfChargeService_.timeWhenChargedOrDepleted().msec()
+                        << "\n" << endl;
```

Comment:
`endl` should already print a new line, so `\n` isn't needed.

Reply:
Also, your format is `Hours: <HH> Min: <MM> Sec: <SS> Ms: <ZZZ>`
It should be `HH:MM:SS.ZZ`